utils/trivial-validate.py

```diff
@@ -42,6 +40,44 @@ def fail(s):
 with open(thispath + '/duplicate-whitelist.txt') as duplicate_fh:
     duplicate_allowed_list = [x.rstrip('\n') for x in duplicate_fh.readlines()]
 
+filenames = glob.glob('src/chrome/content/rules/*')
+
+def generate_hosts_count():
+    # Generate a count of the number of times a host appears, and on which
+    # platforms
+    hosts = {}
```

Comment:
I think you can make this whole function a lot simpler with:

```
c = collections.Counter()
c.update([(hostname, platform)])
for tuple, count in c.most_common():
  if count == 1:
    break
  fail(...)
```

Also, the counting in here should happen within the main loop through the rulesets rather than as a separate step.